src/main/java/org/jenkinsci/plugins/scriptsecurity/sandbox/groovy/SecureGroovyScript.java

```diff
@@ -305,18 +305,29 @@ public Object evaluate(ClassLoader loader, Binding binding) throws Exception {
             
             loader = urlcl = new URLClassLoader(urlList.toArray(new URL[urlList.size()]), loader);
         }
+        boolean canDoCleanup = false;
+
         try {
             loader = GroovySandbox.createSecureClassLoader(loader);
 
-            Field loaderF = GroovyShell.class.getDeclaredField("loader");
-            loaderF.setAccessible(true);
+            Field loaderF = null;
+            try {
+                loaderF = GroovyShell.class.getDeclaredField("loader");
+                loaderF.setAccessible(true);
+                canDoCleanup = true;
```

Comment:
Or simply check `loaderF != null` later; then no `canDoCleanup` variable is needed.